devicemodel/core/sw_load_bzimage.c

```diff
@@ -183,14 +185,6 @@ acrn_prepare_ramdisk(struct vmctx *ctx)
 		return -1;
 	}
 
-	if (len > (BOOTARGS_LOAD_OFF(ctx) - RAMDISK_LOAD_OFF(ctx))) {
-		pr_err("SW_LOAD ERR: the size of ramdisk file is too big"
-				" file len=0x%lx, limit is 0x%lx\n", len,
-				BOOTARGS_LOAD_OFF(ctx) - RAMDISK_LOAD_OFF(ctx));
-		fclose(fp);
-		return -1;
-	}
-
```

Comment:
We need to limit the max size of ramdisk, otherwise it may override the first two sub-regions("ACPI table" and "kernel image"). Also it may met issue when the size exceed the ctx->lowmem.

Reply:
We only have the kernel image and the ACPI table in the memory below RAMDISK_LOAD_OFF. Everything else should be taken care off by the changes in the respective macros. So testing if the kernel image fits into the `RAMDISK_LOAD_OFF - 16*MB` region is sufficient, and this is already tested in `acrn_prepare_kernel()`
https://github.com/projectacrn/acrn-hypervisor/blob/6c02a8d5ed97b248202775244313ec6f53e97293/devicemodel/core/sw_load_bzimage.c#L234-L238
and thus loading with a ramdisk file too large would fail there. This is implicitly the required limitiation of the ramdisk size. Nevertheless we should add a check to verify ramdisk size is small enough to avoid overflows int the calculation macros, i.e. `RAMDISK_LOAD_SIZE < ctx-lowmem - 2*KB`. If you agree, I will prepare an updated patch that contains such a test.